auth/server/src/main/java/com/seagates3/controller/IAMController.java

```diff
@@ -368,25 +376,41 @@ ServerResponse serve(FullHttpRequest httpRequest,
           requestAction.equals("GetTempAuthCredentials") ||
           requestAction.equals("UpdateAccountLoginProfile") ||
           requestAction.equals("DeleteAccount"))) {
-      try {
-        if (RootPermissionAuthorizer.getInstance().containsAction(
-                requestAction)) {
-          new IAMApiAuthorizer().authorizeRootUser(requestor, requestBody);
-        } else {
-          new IAMApiAuthorizer().authorize(requestor, requestBody);
-        }
-        LOGGER.info("Request is authorized for user: " + requestor.getName() +
-                    " account: " + requestor.getAccount());
-      }
-      catch (InvalidUserException e) {
-        LOGGER.error(e.getServerResponse().getResponseBody());
-        return e.getServerResponse();
+      // Authorize with IAM Policy
+      serverResponse = new Authorizer().authorize(requestor, requestBody);
+      if (serverResponse != null &&
+          serverResponse.getResponseStatus() != HttpResponseStatus.OK) {
+        return serverResponse;
       }
     }
 
     return performAction(resourceMap, requestBody, requestor);
   }
-
+  /**
+   * Below will load requestor object with user and policy data using ldap calls
+   * @param requestor
+   */
+ private
+  void loadRequestor(Requestor requestor) {
+    try {
+      requestor.setUser(new UserImpl().findByUserId(
+          requestor.getAccount().getName(), requestor.getId()));
+      List<String> policyDocList = new ArrayList<>();
+      if (!Authorizer.isRootUser(requestor.getUser())) {
+        Map<String, Object> dataMap = new HashMap<>();
+        dataMap.put("accountName", requestor.getUser().getAccountName());
+        dataMap.put("policyIds", requestor.getUser().getPolicyIds());
+        List<Policy> attachedPolicies = new PolicyImpl().findByIds(dataMap);
+        for (Policy policy : attachedPolicies) {
+          policyDocList.add(policy.getPolicyDoc());
+        }
```

Comment:
If user dont have policies attached to it, then dataMap.put("policyIds".......) will be null. 
And in findByIds(), I can see, if policyIds are null then it is returning all the policies in the account. 
And eventually this will lead to populated all the policyDocs in requestor object.
Is this expected ?

Reply:
It is handled. Resolving this one.